htmd/adaptive/adaptive.py

```diff
@@ -203,7 +214,10 @@ def run(self):
                             path.join(self.inputpath, "e" + str(epoch + 1) + "s*")
                         )
                         try:
-                            self.app.submit(natsorted(newsims))
+                            if self.mps > 0:
+                                self._submit_mps(newsims)
+                            else:
+                                self.app.submit(natsorted(newsims))
```

Comment:
just move the if/else inside of the _submit_mps (rename it to just _submit). Cleans up the `run` function.

```
    def _submit(self, newsims):
        from jobqueues.slurmqueue import SlurmQueue
        
        if self.mps == 0:
            self.app.submit(natsorted(newsims))
            return

        if not isinstance(self.app, SlurmQueue):
            raise ValueError(
                "Multiple processes per simulation only supported with SlurmQueue."
            )
        for i in range(0, len(newsims), self.mps):
            batch = newsims[i : min(i + self.mps, len(newsims))]
            self.app.submit(batch, nvidia_mps=len(batch) > 1)
```

Reply:
Done!